File: RepoTransBench/target_projects/Python/Go/tadeck_onetimepass/tests/original/onetimepass_extra_test.go
package original

import (
	"testing"
	"tadeck_onetimepass/onetimepass"
	"crypto/sha256"
	"bytes"
)

func TestGetHOTPInvalidSecretType(t *testing.T) {
	_, err := onetimepass.GetHOTP(1234, 1, onetimepass.HOTPOptions{})
	if err == nil {
		t.Errorf("Expected error for secret of invalid type (int), got nil")
	}
}

func TestGetHOTPIncorrectBase32(t *testing.T) {
	_, err := onetimepass.GetHOTP([]byte("notbase32@#$"), 1, onetimepass.HOTPOptions{})
	if err == nil {
		t.Errorf("GetHOTP should error for non-base32 input")
	}
}

func TestGetHOTPCustomDigest(t *testing.T) {
	val, err := onetimepass.GetHOTP([]byte("MFRGGZDFMZTWQ2LK"), 1, onetimepass.HOTPOptions{DigestMethod: sha256.New, TokenLength: 8})
	if err != nil {
		t.Fatalf("Unexpected error: %v", err)
	}
	// Should return int up to 8 digits
	if _, ok := val.(int); !ok {
		t.Errorf("Expected int HOTP token, got %T", val)
	}
	if l := len(bytes.Trim([]byte(string(val.(int))), " ")); l > 8 {
		t.Errorf("Custom digest HOTP should be at most 8 digits, got len %d", l)
	}
}

func TestGetHOTPStringTypes(t *testing.T) {
	res, err := onetimepass.GetHOTP("MFRGGZDFMZTWQ2LK", 2, onetimepass.HOTPOptions{AsString: true})
	if err != nil {
		t.Fatalf("GetHOTP error: %v", err)
	}
	rb, ok := res.([]byte)
	if ok {
		expect := []byte("816065")
		if !bytes.Equal(rb, expect) {
			t.Errorf("Expected HOTP string token %v, got %v", expect, rb)
		}
	} else {
		t.Errorf("GetHOTP returned type %T, want []byte", res)
	}
}

func TestValidHOTPReturnsFalse(t *testing.T) {
	secret := []byte("MFRGGZDFMZTWQ2LK")
	if ok := onetimepass.ValidHOTP(111111, secret, onetimepass.ValidHOTPOptions{Last: 0, Trials: 1}); ok != false {
		t.Errorf("ValidHOTP on unlikely token should return false")
	}
}

func TestValidHOTPWithRange(t *testing.T) {
	secret := []byte("MFRGGZDFMZTWQ2LK")
	tok, _ := onetimepass.GetHOTP(secret, 99, onetimepass.HOTPOptions{})
	got := onetimepass.ValidHOTP(tok, secret, onetimepass.ValidHOTPOptions{Last: 97, Trials: 3})
	if got != 99 {
		t.Errorf("Expected ValidHOTP(...) == 99, got %v", got)
	}
}

func TestValidTOTPFalse(t *testing.T) {
	secret := []byte("MFRGGZDFMZTWQ2LK")
	if onetimepass.ValidTOTP(123456, secret, onetimepass.ValidTOTPOptions{}) {
		t.Errorf("ValidTOTP should return false for random token")
	}
}

func TestGetTOTPAndValidTOTP(t *testing.T) {
	secret := []byte("MFRGGZDFMZTWQ2LK")
	tok, err := onetimepass.GetTOTP(secret, onetimepass.TOTPOptions{})
	if err != nil {
		t.Fatalf("GetTOTP error: %v", err)
	}
	if !onetimepass.ValidTOTP(tok, secret, onetimepass.ValidTOTPOptions{}) {
		t.Errorf("ValidTOTP(..., correct token) = false, want true")
	}
	if onetimepass.ValidTOTP(tok.(int)+1, secret, onetimepass.ValidTOTPOptions{}) {
		t.Errorf("ValidTOTP with off-by-one should be false")
	}
}

func TestGetTOTPCustomTokenLength(t *testing.T) {
	secret := []byte("MFRGGZDFMZTWQ2LK")
	tok, err := onetimepass.GetTOTP(secret, onetimepass.TOTPOptions{TokenLength: 8})
	if err != nil {
		t.Fatalf("GetTOTP error: %v", err)
	}
	if _, ok := tok.(int); !ok {
		t.Errorf("Expected TOTP int token, got %T", tok)
	}
	if l := len([]byte(string(tok.(int)))); l > 8 {
		t.Errorf("TOTP token should be at most 8 digits, got %d", l)
	}
}

File: RepoTransBench/target_projects/Python/Java/tadeck_onetimepass/src/test/java/com/tadeck/onetimepass/original/OnetimepassExtraTest.java
package com.tadeck.onetimepass.original;

import com.tadeck.onetimepass.Onetimepass;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.security.MessageDigest;

import static org.junit.jupiter.api.Assertions.*;

public class OnetimepassExtraTest {

    private byte[] secret;

    @BeforeEach
    public void setUp() {
        secret = "MFRGGZDFMZTWQ2LK".getBytes();
    }

    @Test
    public void testGetHotpInvalidSecretType() {
        // Pass int instead of bytes - should throw ClassCastException/IllegalArgumentException
        assertThrows(ClassCastException.class, () -> {
            Onetimepass.getHotp((Object) 1234, 1);
        });
    }

    @Test
    public void testGetHotpIncorrectBase32() {
        // Not decodable base32 string: should raise IllegalArgumentException
        assertThrows(IllegalArgumentException.class, () -> {
            Onetimepass.getHotp("notbase32@#$".getBytes(), 1);
        });
    }

    @Test
    public void testGetHotpCustomDigest() throws Exception {
        // Java: Use SHA-256
        MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
        int val = Onetimepass.getHotp(secret, 1, sha256, 8);
        assertTrue(val >= 0);
        assertTrue(Integer.toString(val).length() <= 8);
    }

    @Test
    public void testGetHotpStringTypes() {
        String secretString = "MFRGGZDFMZTWQ2LK";
        Object res = Onetimepass.getHotp(secretString, 2, true);
        assertEquals("816065", res);
    }

    @Test
    public void testValidHotpReturnsFalse() {
        assertFalse(Onetimepass.validHotp(111111, secret, 0, 1));
    }

    @Test
    public void testValidHotpWithRange() {
        int tok = Onetimepass.getHotp(secret, 99);
        assertEquals(99, Onetimepass.validHotp(tok, secret, 97, 3));
    }

    @Test
    public void testValidTotpFalse() {
        assertFalse(Onetimepass.validTotp(123456, secret));
    }

    @Test
    public void testGetTotpAndValidTotp() {
        int tok = Onetimepass.getTotp(secret);
        assertTrue(Onetimepass.validTotp(tok, secret));
        assertFalse(Onetimepass.validTotp(tok + 1, secret));
    }

    @Test
    public void testGetTotpCustomTokenLength() {
        int tok = Onetimepass.getTotp(secret, 8);
        assertTrue(tok >= 0);
        assertTrue(Integer.toString(tok).length() <= 8);
    }
}